### PR TITLE
Add an o1js-compatible modern Mina signature variant

### PR DESCRIPTION
**Pull request by korayakpinar**
## Summary

I added a new Mina Schnorr signature variant that is compatible with the modern o1js `sign()` and `verify()` APIs.

The existing implementation uses the legacy nonce derivation and Legacy Poseidon parameters. Because of that, its signatures work with `verifyLegacy()`, but not with the modern verifier or the in-circuit `Signature.verify()` API.

## What I changed

I kept the existing legacy behavior as-is and added separate constructors for the modern variant:

- `NewModernScheme`
- `NewModernRandomisedScheme`
- `NewModernDeterministicVariant`
- `NewModernRandomisedVariant`

The modern variant:

- uses o1js-compatible chunked input packing,
- derives deterministic nonces using the modern transcript,
- uses Kimchi Poseidon for the signature challenge,
- and treats `devnet` and `testnet` as the same signature domain, matching o1js.

I also added tests for modern signing, deterministic nonce generation, parity correction, legacy/modern separation, and o1js-generated signature vectors.

## Compatibility

I intentionally left `NewScheme()` and the existing legacy constructors unchanged, so current users should continue getting the same signatures as before. The new behavior is only enabled through the new modern constructors.

## How I verified it

Besides the committed Go tests, I ran a local differential test against o1js 2.15.0.

I tested 1,156 field-only cases with randomized values, field and scalar boundary cases, different networks, invalid messages and keys, modified signatures, and both verification directions.

For all positive cases:

- the public keys matched,
- `r` and `s` matched,
- the serialized 64-byte signatures matched byte-for-byte,
- Go signatures passed o1js verification,
- and o1js signatures passed Go verification.

I also compiled a small o1js `ZkProgram` and verified a representative signature inside the circuit.

## Checks

```bash
GOCACHE=/tmp/go-build-cache go test -count=1 -tags=purego ./pkg/signatures/schnorrlike/mina
GOCACHE=/tmp/go-build-cache go vet -tags=purego ./pkg/signatures/schnorrlike/mina

## Description

## Pull Request Checklist

### Scope
- [x] Summary and description are provided.
- [x] Changes are focused and scoped to one purpose.

### Testing (select all that apply)
- [x] Unit tests
- [ ] Property tests
- [ ] Test vectors tests
- [ ] Benchmarks
- [ ] Not run (explain why)

### Documentation / Spec (if relevant)
- [ ] Updated/added docs or comments
- [ ] Spec updated or linked

### Notes
- [x] Additional context is provided (if needed)

### CODE REVIEW

File: pkg/signatures/schnorrlike/mina/mina.go [hunk 1]
```diff
@@ -36,9 +36,10 @@ type (
 )
 
 var (
-	hashFunc = hashing.HashFuncTypeErase(poseidon.NewLegacy)
-	group    = pasta.NewPallasCurve()
-	sf       = pasta.NewPallasScalarField()
+	hashFunc       = hashing.HashFuncTypeErase(poseidon.NewLegacy)
+	modernHashFunc = hashing.HashFuncTypeErase(poseidon.NewKimchi)
```
Comment (mkramarczyk) [inline pkg/signatures/schnorrlike/mina/mina.go]: I don't like usage of "modern" in it.
Let's stick to the "legacy" and "kimchi", or "legacy" and "" (nothing).
so this would become: `legacyHashFunc` and `hashFunc` (or kimchiHashFunc)

Similarly for scheme:
"NewLegacyScheme", "NewKimchiScheme" and so one.
same for variant

**Comment by mkramarczyk** [comment]
@korayakpinar mostly looks good to me, please address this "modern" wording (instead use no prefix for what now is modern and use "legacy" prefix for, well, legacy. Rebase and it would be good to merge.

**Review by mkramarczyk — CHANGES_REQUESTED** [review]
Please rething using "modern" prefix. I don't think modern is a proper wording in Mina context (or at least I don't recall mina creator use modern anywhere, it might be confusing).
Otherwise it looks good to me.

File: pkg/signatures/schnorrlike/mina/mina.go (hunk 1 above)
Comment (korayakpinar) [reply pkg/signatures/schnorrlike/mina/mina.go]: Done.

**Comment by korayakpinar** [comment]
@mkramarczyk Friendly bump!

**Review by mkramarczyk — APPROVED** [review]
looks good to me
